rust/segment/src/sqlite_metadata.rs

```diff
@@ -778,6 +858,8 @@ impl SqliteMetadataReader {
 
 #[cfg(test)]
 mod tests {
```

Comment:
Suggest adding a few coverage tests for common edge cases

- Metadata key null for one record, exists for another
- FTS match but no metadata match